src/ledger/P23HotArchiveBug.cpp

```diff
@@ -451,5 +451,211 @@ Protocol23CorruptionDataVerifier::verifyEntryFixesOnP24Upgrade(
     }
 }
 
+Protocol23CorruptionEventReconciler::Protocol23CorruptionEventReconciler(
+    Hash const& networkID)
+{
+    for (size_t i = 0; i < P23_CORRUPTED_AFFECTED_ASSETS_COUNT; ++i)
+    {
+        Asset asset;
+        fromOpaqueBase64(asset, P23_CORRUPTED_AFFECTED_ASSETS[i]);
+
+        SCAddress address;
+        address.type(SC_ADDRESS_TYPE_CONTRACT);
+        address.contractId() = getAssetContractID(networkID, asset);
+
+        auto [_i, inserted] = mSACAssetMap.emplace(address, asset);
+        releaseAssert(inserted);
+    }
+
+    for (size_t i = 0; i < P23_CORRUPTED_HOT_ARCHIVE_ENTRIES_COUNT; ++i)
+    {
+        LedgerEntry corruptedEntry =
+            decodeLedgerEntry(P23_CORRUPTED_HOT_ARCHIVE_ENTRIES[i]);
+
+        LedgerEntry correctEntry =
+            decodeLedgerEntry(P23_CORRUPTED_HOT_ARCHIVE_ENTRY_CORRECT_STATE[i]);
+
+        mKeyToEntries.emplace(LedgerEntryKey(corruptedEntry),
+                              std::make_pair(correctEntry, corruptedEntry));
+    }
+}
+
+std::optional<std::pair<uint64_t /*lo balance*/, SCAddress /*owner*/>>
+getAmountLo(LedgerEntry const& le)
```

Comment:
nit: This name is a bit confusing, maybe call it something like just `getSACBalance`? it's implied that SAC balance is i64.

Reply:
I'm cleaning this up as I know none of the relevant balances exceed int64_t now, but to be clear, it's possible for a contracts SAC balance to exceed int64max. I didn't make that assumption earlier so I could log any balances that did (if there were any).